server/src/main/java/com/hedera/block/server/persistence/storage/PersistenceStorageConfig.java

```diff
@@ -56,6 +63,7 @@ public record PersistenceStorageConfig(
      */
     public PersistenceStorageConfig {
         Objects.requireNonNull(type);
+        Objects.requireNonNull(compression).verifyCompressionLevel(compressionLevel);
```

Comment:
We do not need the first requireNonNull; the call will throw without it.
```suggestion
        compression.verifyCompressionLevel(compressionLevel);
```
General principle: Use `requireNonNull` if we would not otherwise throw a NPE within a constructor, or within the first few lines of any other method.

Letting the "organic" NPE throw is preferable because the JVM will usually fill in better details than `Objects.requireNonNull` will.

Reply:
✅ Done.